File: app/src/androidTest/java/com/test/demo/PhoneNumTest.java
package com.test.demo;

import android.Manifest;
import android.content.Intent;
import android.net.Uri;
import android.support.test.espresso.intent.rule.IntentsTestRule;
import android.support.test.rule.GrantPermissionRule;
import android.support.test.runner.AndroidJUnit4;
import android.test.suitebuilder.annotation.MediumTest;

import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.action.ViewActions.click;
import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
import static android.support.test.espresso.action.ViewActions.typeText;
import static android.support.test.espresso.intent.Intents.intended;
import static android.support.test.espresso.intent.matcher.IntentMatchers.hasAction;
import static android.support.test.espresso.intent.matcher.IntentMatchers.hasData;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static org.hamcrest.Matchers.allOf;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.core.IsEqual.equalTo;


@RunWith(AndroidJUnit4.class)
@MediumTest
public class PhoneNumTest {

    @Rule
    public IntentsTestRule<PhoneActivity> mActivityRule = new IntentsTestRule<>(PhoneActivity.class);

    @Rule
    public GrantPermissionRule permissionRule = GrantPermissionRule.grant(Manifest.permission.CALL_PHONE);

    public String VALID_PHONE_NUMBER;
    public Uri INTENT_DATA_PHONE_NUMBER;

    @Before
    public void setUp() throws Exception {
        VALID_PHONE_NUMBER = "10085";
        INTENT_DATA_PHONE_NUMBER = Uri.parse("tel:" + VALID_PHONE_NUMBER);
    }

    @Test
    public void callPhoneTest() {
        onView(withId(R.id.etPhone)).perform(typeText(VALID_PHONE_NUMBER),
            closeSoftKeyboard());
        onView(withId(R.id.btnCall)).perform(click());

        intended(allOf(
            hasAction(is(equalTo(Intent.ACTION_CALL))),
            hasData(equalTo(INTENT_DATA_PHONE_NUMBER))
        ));
    }



}

File: app/src/androidTest/java/com/test/demo/PhoneNumTest.kt
package com.test.demo

import android.Manifest
import android.support.test.espresso.Espresso.onView
import android.support.test.espresso.action.ViewActions.click
import android.support.test.espresso.intent.Intents.intended
import android.support.test.espresso.intent.matcher.ComponentNameMatchers.hasShortClassName
import android.support.test.espresso.intent.matcher.IntentMatchers.hasComponent
import android.support.test.espresso.intent.rule.IntentsTestRule
import android.support.test.espresso.matcher.ViewMatchers.withId
import android.support.test.rule.GrantPermissionRule
import android.support.test.runner.AndroidJUnit4
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.hamcrest.Matchers.notNullValue
import org.junit.Before
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith


@RunWith(AndroidJUnit4::class)
class PhoneNumTest {

    @Rule
    @JvmField
    var mActivityRule = IntentsTestRule(PhoneActivity::class.java)

    @Rule
    @JvmField
    var permissionRule = GrantPermissionRule.grant(Manifest.permission.CALL_PHONE)

//    var VALID_PHONE_NUMBER: String
//    var INTENT_DATA_PHONE_NUMBER: Uri

    @Before
    fun setUp() {

//        VALID_PHONE_NUMBER = "10085"
//        INTENT_DATA_PHONE_NUMBER = Uri.parse("tel:$VALID_PHONE_NUMBER")
    }

    fun callPhoneTest() {
//        onView(withId(R.id.etPhone)).perform(typeText(VALID_PHONE_NUMBER),
//                closeSoftKeyboard())
//        onView(withId(R.id.btnCall)).perform(click())
//
//        intended(allOf(
//                hasAction(`is`(equalTo(Intent.ACTION_CALL))),
//                hasData(equalTo(INTENT_DATA_PHONE_NUMBER))
//        ))
    }


    @Test
    fun testJump() {
        val applicationContext = (mActivityRule.activity.application as TestApplication).applicationContext

        assertThat(applicationContext, `is`(notNullValue()))
        //        mActivityRule.launchActivity(null);
        onView(withId(R.id.jumpTwo)).perform(click())
        intended(hasComponent(hasShortClassName(".TwoActivity")))
        //        mActivityRule.launchActivity(null);
    }


}
